### PR TITLE
remove public preview warnings for privatelink connections

### PR DESCRIPTION
**Pull request by jubrad**
### Motivation
The tasks to promote to GA are done, time to remove the labels. Figure I'd get the ball rolling here.
https://github.com/MaterializeInc/materialize/issues/25503

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: doc/user/content/ingest-data/mysql/amazon-rds.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ---
+{{< public-preview />}}
+
```
Comment (benesch) [inline doc/user/content/ingest-data/mysql/amazon-rds.md:3]: I don't think this works in this position!

**Review by benesch — APPROVED** [review]
LGTM, modulo the errant tag in content/ingest-data/mysql/amazon-rds.md.

**Comment by benesch** [comment]
I'm so excited for this!

**Review by pH14 — APPROVED** [review]
😍

**Review by morsapaes — APPROVED** [review]
Rad. Thank you, @jubrad!

**Pull request merged**